lib/app/utils/validation.rb

```diff
@@ -192,5 +192,13 @@ def self.guess_r4_profile(resource)
     def self.observation_contains_code(observation_resource, code)
       observation_resource&.code&.coding&.any? { |coding| coding&.code == code }
     end
+
+    def self.resource_contains_category(resource, category_code, category_system = nil)
+      resource&.category&.any? do |category|
+        category.coding&.any? do |coding|
+          coding.code == category_code && (category_system.blank? || category_system == coding.system)
```

Comment:
If the system leaves the `system` blank but fills in the `code`, is that ok?  This obviously requires the system to be filled in otherwise it won't match.

Reply:
good catch!

Reply:
fixed